src/core/open-file.c

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include "open-file.h"
+#include "string-util.h"
+#include "strv.h"
+
+int parse_open_file(const char *v, OpenFile *ret) {
+    _cleanup_strv_free_ char **parts = NULL;
+    int r;
+
+    if (!v)
+        return -EINVAL;
+
+    r = strv_split_full(&parts, v, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+    if (r < 0)
+            return r;
+    if (strv_length(parts) != 3)
+            return -EINVAL;
+
+    ret->path = strdup(parts[0]);
+    ret->fdname = strdup(parts[1]);
+    ret->flags = streq(parts[2], "rw") ? O_RDWR : O_RDONLY;
+    ret->fd = -1;
+
+    return 0;
+}
+
+const char *open_file_to_string(const OpenFile *open_file) {
+    if (!open_file)
+        return "";
+    return strjoin(open_file->path, ":", open_file->fdname, ":", open_file->flags == O_RDWR ? "rw" : "ro");
+}
+
+void free_open_file_fields(const OpenFile *open_file) {
```

Comment:
This should be named `open_file_free()` and it should free both the members of `open_file` and `open_file` itself as well.